### PR TITLE
gh release action

### PR DESCRIPTION
**Pull request by knaaptime**
this starts a new gh action that should do the folowing

upon the push of a new tag prefixed with "v":

- build the package and upload to pypi
- run the changelog notebook
- create a new github release and upload the package asset

### CODE REVIEW

**Comment by knaaptime** [comment]
i think this is ready, but not exactly sure how to test?

**Comment by jGaboardi** [comment]
Maybe try a post release, i.e. `v0.2.0.post1`?

**Comment by knaaptime** [comment]
@jGaboardi  ha i was just searching for your handle to tag you as a reviewer. So just push a new tag `v0.2.0.post1` should work? not touch the version in the init or anything?

**Comment by jGaboardi** [comment]
hmmm. We are both in new territory here... Yeah, maybe we could try doing a empty commit named `v0.2.0.post1`, but my intuition says we would need to (should) actually change the version in `__init__`. If not for actual functionality, then to preserve the state of `geosnap` at `v0.2.0` as pre-GHA releases.

File: .github/workflows/upload_package.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install setuptools wheel twine jupyter
+      run: |
+        python setup.py sdist bdist_wheel
+    - name: Publish distribution 📦 to PyPI
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        password: ${{ secrets.pypi_password }}
```
Comment (jGaboardi) [inline .github/workflows/upload_package.yml:28]: just curious for future reference: you are storing your PyPI password as a GH secret in the `geosnap` repo?

Comment (knaaptime) [reply .github/workflows/upload_package.yml:28]: its an api key you create on pypi, then enter into the repository's settings. I followed https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/

**Comment by knaaptime** [comment]
post release is a good idea. maybe even a `0.2.1`

**Comment by knaaptime** [comment]
what im worried about is, if it still needs fixing, i dont want to keep incrementing versions just to push a tag

**Comment by jGaboardi** [comment]
> what im worried about is, if it still needs fixing, i dont want to keep incrementing versions just to push a tag

I think that's the benefit of the `v0.2.0.post#` structure. Keep incrementing up until we have the workflow correct, then bump to `v0.2.1`.

**Comment by knaaptime** [comment]
ah, got it.

**Comment by jGaboardi** [comment]
Or even better: start with a `v0.2.1.rc1` tag and increment up until it is all good then drop the `.rc#` tag

**Comment by knaaptime** [comment]
cool, this is working. only took 14 tries but here are the [releases](https://github.com/knaaptime/geosnap/releases) its been making on my account and the rcs are up [on pypi](https://pypi.org/project/geosnap/#history)

File: README.md [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 <h2 align="center" style="margin-top:-10px">The Geospatial Neighborhood Analysis Package</h2> 
 
-[![Build Status](https://travis-ci.com/spatialucr/geosnap.svg?branch=master)](https://travis-ci.com/spatialucr/geosnap)
+![.github/workflows/unittests.yml](https://github.com/knaaptime/geosnap/workflows/.github/workflows/unittests.yml/badge.svg)
```
Comment (jGaboardi) [inline README.md:7]: When clicked, this badge only takes you to an image of the badge, not the test result itself. I had quite a time getting it correct in `spaghetti`. Here is what works for that (from [`README.md`](https://github.com/pysal/spaghetti/blob/master/README.md#spatial-graphs-networks-topology--inference)):

```
[![unittests](https://github.com/pysal/spaghetti/workflows/.github/workflows/unittests.yml/badge.svg)](https://github.com/pysal/spaghetti/actions?query=workflow%3A.github%2Fworkflows%2Funittests.yml)
```

Comment (knaaptime) [reply README.md]: nice, thanks

**Comment by knaaptime** [comment]
this is throwing an unknown error, but [passing](https://github.com/knaaptime/geosnap/runs/612462634?check_suite_focus=true) on my fork, so i think i'd like to merge this, then do the bump to 0.2.1 in a second PR (any substantive changes need to be made prior to the release commit, otherwise they wont show in the changelog)

**Comment by jGaboardi** [comment]
pysal/submodule_template#28

**Pull request merged**